flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/common/CommonTemporalTableJoinRule.scala

```diff
@@ -44,20 +44,15 @@ trait CommonTemporalTableJoinRule {
 
     // period specification check
     snapshot.getPeriod match {
-      // it's left table's field, pass
-      case r: RexFieldAccess if r.getReferenceExpr.isInstanceOf[RexCorrelVariable] =>
+      // it should be left table's field and is a time attribute
+      case r: RexFieldAccess
+        if r.getType.isInstanceOf[TimeIndicatorRelDataType] &&
+          r.getReferenceExpr.isInstanceOf[RexCorrelVariable] => // pass
       case _ =>
         throw new TableException("Temporal table join currently only supports " +
-          "'FOR SYSTEM_TIME AS OF' left table's time attribute field, doesn't support 'PROCTIME()'")
+          "'FOR SYSTEM_TIME AS OF' left table's time attribute field.")
     }
 
-    snapshot.getPeriod.getType match {
-      // supports both event-time and processing time
-      case t: TimeIndicatorRelDataType =>
-      case _ =>
-        throw new TableException("Temporal table join currently only supports " +
-          "'FOR SYSTEM_TIME AS OF' left table's time attribute field")
-    }
```

Comment:
Do we check the period type in other place ?

Reply:
I merged it into the above check.